src/main/groovy/info/solidsoft/gradle/pitest/report/PitestReportAggregator.groovy

```diff
@@ -0,0 +1,111 @@
+package info.solidsoft.gradle.pitest.report
+
+import groovy.transform.CompileStatic
+import info.solidsoft.gradle.pitest.PitestPlugin
+import info.solidsoft.gradle.pitest.PitestPluginExtension
+import info.solidsoft.gradle.pitest.PitestTask
+import org.gradle.api.Project
+import org.gradle.api.UnknownDomainObjectException
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.RegularFile
+import org.gradle.api.provider.Provider
+import org.pitest.aggregate.ReportAggregator
+import org.pitest.mutationtest.config.DirectoryResultOutputStrategy
+import org.pitest.mutationtest.config.UndatedReportDirCreationStrategy
+
+import java.nio.file.Paths
+
+@CompileStatic
+class PitestReportAggregator {
+
+    private static final String MUTATION_FILE_NAME = "mutations.xml";
+    private static final String LINECOVERAGE_FILE_NAME = "linecoverage.xml";
+
+    void aggregate(Project root) {
+        Set<Project> subprojects = getSubprojectsWithPitest(root)
+        boolean rootContainPitest = isRootPitestConfigured(root)
+        if (rootContainPitest || !subprojects.isEmpty()) {
+            root.getLogger().info("Aggregating pitest reports")
+            ReportAggregator.Builder builder = ReportAggregator.builder()
+            if (rootContainPitest) {
+                addProjectFiles(builder, root)
+            }
+            subprojects.each { project ->
+                addProjectFiles(builder, project)
+            }
+
+            String outputDir = Paths.get(root.buildDir.absolutePath, "reports", "pitest").toFile().absolutePath
```

Comment:
In the single report configuration (in `PitestPlugin`) I use:
```
        extension.reportDir.set(new File(project.extensions.getByType(ReportingExtension).baseDir, "pitest"))
```
Would it be worth to use that construction also here? Probably "pitest"` could be extracted to a constant in `PitestPlugin`.

Reply:
This change requires root plugin to contains JavaPlugin. I wrote some workaround code, check it please